### ISSUE TITLE
Got an error [NPE] upon clicking the Shared section: `OCFileListAdapter.getFooterText`

### ISSUE THREAD

**Issue by TheRedXD** [nextcloud/android#11924]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

I haven't found the specific steps to reproduce, the app just crashed upon clicking Shared section iirc

### Expected behaviour

The app shouldn't have crashed

### Actual behaviour

Upon clicking the Shared button, the app froze for like 2 seconds and threw me an error (it crashed)

### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.owncloud.android.datamodel.OCFile.isFolder()' on a null object reference
    at com.owncloud.android.ui.adapter.OCFileListAdapter.getFooterText(OCFileListAdapter.java:540)
    at com.owncloud.android.ui.adapter.OCFileListAdapter.onBindViewHolder(OCFileListAdapter.java:361)
    at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7678)
    at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7761)
    at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6582)
    at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6848)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6688)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6684)
    at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4604)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4307)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4873)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:689)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1818)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1584)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
    at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
    at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1263)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at com.android.internal.policy.DecorView.onLayout(DecorView.java:1418)
    at android.view.View.layout(View.java:24958)
    at android.view.ViewGroup.layout(ViewGroup.java:6784)
    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4787)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4115)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2991)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10665)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1301)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1309)
    at android.view.Choreographer.doCallbacks(Choreographer.java:923)
    at android.view.Choreographer.doFrame(Choreographer.java:852)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1283)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8741)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30250090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a53x`
* Model: `SM-A536B`
* Id: `TP1A.220624.014`
* Product: `a53xnaeea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `A536BXXS4BWA2`

### Android version

13

### Device brand and model

Samsung Galaxy A53

### Stock or custom OS?

Stock

### Nextcloud android app version

3.25.0

### Nextcloud server version

Whatever is the latest in the nextcloud docker hub package, I can't look at it right now

### Using a reverse proxy?

Yes

**Comment by NL-TCH** [comment]
Got the same:

```
### Cause of error
```java
Exception in thread "main" java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 3(offset:3).state:23 com.owncloud.android.ui.EmptyRecyclerView{6442b6a VFED..... ......ID 0,0-720,1340 #7f0a02ab app:id/list_root}, adapter:com.owncloud.android.ui.adapter.OCFileListAdapter@adc6e5c, layout:androidx.recyclerview.widget.LinearLayoutManager@56b5987, context:com.owncloud.android.ui.activity.FileDisplayActivity@78b275a
    at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6752)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6688)
    at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6684)
    at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
    at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
    at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
    at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4604)
    at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4307)
    at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4873)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:689)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1818)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1584)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
    at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
    at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1263)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
    at com.android.internal.policy.DecorView.onLayout(DecorView.java:1227)
    at android.view.View.layout(View.java:24475)
    at android.view.ViewGroup.layout(ViewGroup.java:7383)
    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4260)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3695)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2618)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9971)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1010)
    at android.view.Choreographer.doCallbacks(Choreographer.java:809)
    at android.view.Choreographer.doFrame(Choreographer.java:744)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:246)
    at android.app.ActivityThread.main(ActivityThread.java:8653)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30250090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a20e`
* Model: `SM-A202F`
* Id: `RP1A.200720.012`
* Product: `a20eeea`

### Firmware
* SDK: `30`
* Release: `11`
* Incremental: `A202FXXU5CWF2`
```

**Comment by Jarige** [comment]
I think I've got the same error. I think the problem is that Nextcloud tries to open a folder because that folder was the last place the user uploaded a file. But when that folder no longer exists, it causes this problem.

**Comment by joshtrichards** [comment]
@NL-TCH Your matter is unrelated to the reporter's issue. Though your situation is fixed in #11804 which was released in v3.26.0

**Comment by joshtrichards** [comment]
This issue may have been addressed in v3.26.0 of the client, but can't say with certainty. Unfortunate if it lacks reproduction steps it's near impossible to verify.
